plugins/searchonline.koplugin/handleurlstring.lua

```diff
@@ -0,0 +1,171 @@
+local InfoMessage = require("ui/widget/infomessage")
+local UIManager = require("ui/uimanager")
+local url = require("socket.url")
+local util = require("util")
+local _ = require("gettext")
+
+local HandleUrlString = {}
+
+-- Message box to inform the user about a misformatted URL
+function HandleUrlString:urlError(text_explaining_error)
+    UIManager:show(InfoMessage:new{
+        text = _(text_explaining_error),
+        timeout = 3
+    })
+end
+
+-- Check if the URL is valid and show an error if not
+function HandleUrlString:isValidUrl(url_string)
+    local parsed_url = url.parse(url_string)
+
+    if not parsed_url then
+        self.urlError("Could not parse URL. It may be misformatted: " .. url_string)
+        return false
+    end
+
+    if not parsed_url.scheme then
+        self.urlError("The URL is missing http:/https: " .. url_string)
+        return false
+    end
+
+    if not parsed_url.host then
+        self.urlError("Missing or invalid host: " .. url_string)

```

Comment:
I'm not sure all these distinctive cases needs to be that inspected. I think nowhere we have such scrutiny and that many error messages to propose to translators (it's here you would add `_(...)`).
"Invalid url" would be enough, even browsers don't do that much :)

Reply:
We can't highlight/bold stuff in our text widgets (unless wrapping bits with chinese chevrons...)

I understand you'd like to be clear - I'm just saying that we nowhere are going into such depths of analysis and details. It's just much code to maintain for not much gain.
I would probably just check for the presence of your specific {q}, and let the rest be checked by the web browsers (and let it tell the user why it can process the url :))